### PR TITLE
Updated angular-step-by-step document

### PR DESCRIPTION
**Pull request by alirizaadiyahsi**

### CODE REVIEW

File: doc/Getting-Started-Core.html [hunk 1]
```diff
@@ -37,18 +40,37 @@ <h3 id="DocCreateProject">Create a Project</h3>
 
 <h3 id="DocConfigureProject">Configure The Project</h3>
 
+
+<div class="alert alert-warning">
+    <strong>Important Notice:</strong><br>
+    <p>Installing client side npm dependencies using <strong>yarn</strong> before opening the solution will increase project opening &amp; building time dramatically.</p>
```
Comment (acjh) [inline doc/Getting-Started-Core.html]: Should this be `decrease`?

File: doc/Getting-Started-Angular.html [hunk 2]
```diff
@@ -141,12 +143,12 @@ <h4>Application UI</h4>
...
 see the sample dashboard screen:</p>
 <p>
-<img class="img-thumbnail" alt="Dashboard" height="965" src="images/dashboardV2.png" width="1235" /></p>
+<img class="img-thumbnail" alt="Dashboard" height="965" src="images/dashboardV3.png" width="1235" /></p>
 <h3 id="DocMore">More</h3>
 
 <p>Your solution is up and working.
 See
-<a href="Development-Guide-Angular.html"><span class="text-primary">development guide</span></a> document for more 
+<a href="Getting-Started-Xamarin.html"><span class="text-primary">getting started xamarin</span></a> for xamarin application, <a href="Development-Guide-Angular.html"><span class="text-primary">development guide</span></a> document for more
```
Comment (acjh) [inline doc/Getting-Started-Angular.html]: Capital `X` for `Xamarin`? (also: [here](https://github.com/aspnetzero/documents/pull/29/files#diff-4da194f25b0fea35173718c38fbbd369R124))

File: doc/Getting-Started-Angular.html [hunk 3]
```diff
@@ -42,10 +42,12 @@ <h4>Merging Client and Server Solutions</h4>
...
 </p>
 <h3>ASP.NET Core Application</h3>
-<p>When you open Server side solution in <strong>Visual Studio 2017+</strong>, you will see the 
+<p>When you open Server side solution
+<strong style="box-sizing: border-box; font-weight: 700; color: rgb(51, 51, 51); font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-style: initial; text-decoration-color: initial;">
+*.Web.sln</strong> in <strong>Visual Studio 2017+</strong>, you will see the 
 solution structure as like below:</p>
 <p>
-<img class="img-thumbnail" alt="AspNet Core solution structure" src="images/aspnet-core-host-solution-3.png" /></p>
+<img class="img-thumbnail" alt="AspNet Core solution structure" src="images/aspnet-core-host-solution-4.png" /></p>
```
Comment (acjh) [inline doc/Getting-Started-Angular.html]: `ASP.NET Core`?

File: doc/Getting-Started-Core.html (hunk 1 above)
Comment (ismcagdas) [reply doc/Getting-Started-Core.html]: Thanks @acjh , you are great 😄 . Will fix it with next commit.

File: doc/Getting-Started-Core.html [hunk 4]
```diff
@@ -37,18 +40,37 @@ <h3 id="DocCreateProject">Create a Project</h3>
...
+
+<h3 id="DocPrerequirements">Prerequirements</h3>
+
+<p>MVC application needs to following tools be installed:</p>
+
+<ul>
+	<li><a href="https://nodejs.org/en/download/" target="_blank">nodejs</a> 
+	6.9+ with npm 3.10+</li>
+	<li><a href="https://www.npmjs.com/package/gulp">gulp (must be installed globally)</a></li>
+	<li><a href="https://yarnpkg.com/">yarn</a></li>
```
Comment (acjh) [inline doc/Getting-Started-Core.html]: Standardise behavior for `target="_blank"`

File: doc/Getting-Started-Core.html [hunk 5]
```diff
@@ -76,29 +98,30 @@ <h4 id="DocConfigureMultiTenancy">Multi-Tenancy</h4>
...
 see the sample dashboard screen:</p>
 <p>
-<img class="img-thumbnail" alt="Dashboard" height="965" src="images/dashboardV2.png" width="1235" /></p>
+<img class="img-thumbnail" alt="Dashboard" height="965" src="images/dashboardV3.png" width="1235" /></p>
 <h3 id="DocMore">More</h3>
 
 <p>Your solution is up and working.
 See
-<a href="Development-Guide-Core.html"><span class="text-primary">development guide</span></a> document for more 
+<a href="Getting-Started-Xamarin.html"><span class="text-primary">getting started xamarin</span></a> for xamarin application, <a href="Development-Guide-Core.html"><span class="text-primary">development guide</span></a> document for more
```
Comment (acjh) [inline doc/Getting-Started-Core.html]: Requires same changes as https://github.com/aspnetzero/documents/pull/29/commits/23bc6cef6615581f22ce09d893baae8f71797d51#diff-168d6e075db031331a6ae09395f1ae97L151

File: doc/Getting-Started-Core.html [hunk 6]
```diff
@@ -37,18 +40,37 @@ <h3 id="DocCreateProject">Create a Project</h3>
...
+<div class="alert alert-warning">
+    <strong>Important Notice:</strong><br>
+    <p>Installing client side npm dependencies using <strong>yarn</strong> before opening the solution will decrease project opening &amp; building time dramatically.</p>
+</div>
+
+<p>Before opening the solution open a command prompt, navigate to root directory of <strong>*.Web.Mvc</strong> project and run "<strong>yarn</strong>" command to install client side dependencies.</p>
+
+<h3 id="DocPrerequirements">Prerequirements</h3>
+
+<p>MVC application needs to following tools be installed:</p>
```
Comment (acjh) [inline doc/Getting-Started-Core.html]: "needs `the` following tools `to` be installed"

File: doc/Development-Guide-Xamarin.html [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+create native cross platform user interface layouts.<br>
+This document is intended to show how to setup a Xamarin development environment 
+and introduce existing features, user interfaces on Asp.Net Zero Xamarin platform.<br>
+For an overview of the Xamarin installation and setup practices see
+<a href="https://developer.xamarin.com/guides/xamarin-forms/getting-started/installation/">
+Xamarin.Forms Requirements</a> and
+<a href="https://developer.xamarin.com/guides/cross-platform/getting_started/installation/">
+Installation</a>. <br>
+For an overview of the Asp.Net Zero getting started see
+<a href="/Getting-Started-Core">Asp.Net Zero Getting Started.</a> The Asp.Net Zero
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: `"Getting-Started-Core"` should be without the slash

File: doc/Development-Guide-Xamarin.html [hunk 8]
```diff
@@ -0,0 +1,554 @@
...
+For an overview of the Xamarin installation and setup practices see
+<a href="https://developer.xamarin.com/guides/xamarin-forms/getting-started/installation/">
+Xamarin.Forms Requirements</a> and
+<a href="https://developer.xamarin.com/guides/cross-platform/getting_started/installation/">
+Installation</a>. <br>
+For an overview of the Asp.Net Zero getting started see
+<a href="/Getting-Started-Core">Asp.Net Zero Getting Started.</a> The Asp.Net Zero 
+Xamarin is only available for Asp.Net Core! (Angular UI or Mvc UI)</p>
+<h4>Prerequisites</h4>
+<p>Following tools are needed in order to develop ASP.NET Zero Xamarin:</p>
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: - Standardise `Asp.Net Zero` or `ASP.NET Zero`? ([some documents](https://github.com/aspnetzero/documents/pull/29/files#diff-168d6e075db031331a6ae09395f1ae97R69) use `AspNet Zero` too)
- But should standardise `ASP.NET Core` (e.g. [line 28](https://github.com/aspnetzero/documents/pull/29/files#diff-f1999aae9179dae277e5d40ab4bd3410R28))

File: doc/Development-Guide-Xamarin.html [hunk 9]
```diff
@@ -0,0 +1,560 @@
...
+<ul>
+	<li><a href="https://www.visualstudio.com">Visual Studio 2017 v15.4.4</a>+</li>
+	<li>MAC for IOS simulator</li>
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: "`Mac` for `iOS` simulator" (and many other places that use `IOS`)

File: doc/Development-Guide-Xamarin.html [hunk 10]
```diff
@@ -0,0 +1,560 @@
...
+<p>You can open <b>*.Mobile.sln</b> for Xamarin development.&nbsp; </p>
+<table width="100%" cellspacing="1">
+	<tr>
+		<td width="228">
+		<img class="img-thumbnail" alt="ASP.NET Zero Xamarin Architecture" src="images/xamarin-architecture.png" /></td>
+		<td valign="top">
+		<img class="img-thumbnail" alt="ASP.NET Zero Mobile Solution" src="images/mobile-solution-overview.png" /><p>&nbsp;</p>
+		<p>&nbsp;There are 6 projects in the mobile solution:</p>
+		<ul>
+			<li><strong>Core.Shared</strong> project basic <span lang="EN-US">primitive</span><span lang="EN-US" style="font-size: 12.0pt; line-height: 107%; font-family: 'Times New Roman',serif">
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: "project `contains` basic"

File: doc/Development-Guide-Xamarin.html [hunk 11]
```diff
@@ -0,0 +1,560 @@
...
+	information read
+	<a href="https://developer.xamarin.com/guides/cross-platform/live/limitations/">
+	Xamarin Live Player limitations.</a></p>
+	<img class="img-thumbnail" alt="Debug IP Address" src="images/xamarin-api-url-config.png" />
+</div>
+<div class="bs-callout bs-callout-info">
+	<p><i class="fa fa-info-circle"></i><a href="https://www.livexaml.com/">LiveXAML</a> 
+	is a live Xaml previewer runs while you are debugging your application inside 
+	an emulator. Whenever you save any XAML file, it automatically updates the running 
+	application. If you want to purchase LiveXAML, Asp.Net Zero customers get %30
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: `30%`

File: doc/Development-Guide-Xamarin.html [hunk 12]
```diff
@@ -0,0 +1,560 @@
...
+			Abp module (in AbpZeroTemplateXamarinAndroidModule class). As this app 
+			requires internet connection, first thing is checking whether device 
+			can connect to internet or not. If there&#39;s no internet connection, a 
+			confirmation dialog appears on the device to ask user enable internet. 
+			SplashActivity routes to MainActivity</li>
+			<li><strong>MainActivity </strong>This is the main startup Activity 
+			for the application. In this acitivty, platform specific intializations 
+			are done. Unobserved and Unhandled exceptions handlers here with ExceptionHandler 
+			class. MainActivity routes to App class which is in the shared project.</li>
+			<li><strong>CropViewRenderer</strong> Is a platform specific renderer
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: Standardise for `This is`

File: doc/Development-Guide-Xamarin.html [hunk 13]
```diff
@@ -0,0 +1,560 @@
...
+			operations while starting up like bootstrapping ABP, retrieving configuration 
+			from host. Android does not have default way to show a splash screen 
+			on startup. SplashActivity class is the main launcher. It initializes 
+			Abp module (in AbpZeroTemplateXamarinAndroidModule class). As this app 
+			requires internet connection, first thing is checking whether device 
+			can connect to internet or not. If there&#39;s no internet connection, a 
+			confirmation dialog appears on the device to ask user enable internet. 
+			SplashActivity routes to MainActivity</li>
+			<li><strong>MainActivity </strong>This is the main startup Activity 
+			for the application. In this acitivty, platform specific intializations
```
Comment (acjh) [inline doc/Development-Guide-Xamarin.html]: `initializations`

File: doc/Getting-Started-Angular.html [hunk 14]
```diff
@@ -141,12 +147,12 @@ <h4>Application UI</h4>
...
 see the sample dashboard screen:</p>
 <p>
-<img class="img-thumbnail" alt="Dashboard" height="965" src="images/dashboardV2.png" width="1235" /></p>
+<img class="img-thumbnail" alt="Dashboard" height="965" src="images/dashboardV3.png" width="1235" /></p>
 <h3 id="DocMore">More</h3>
 
 <p>Your solution is up and working.
 See
-<a href="Development-Guide-Angular.html"><span class="text-primary">development guide</span></a> document for more 
+<a href="Getting-Started-Xamarin.html"><span class="text-primary">getting started</span></a> for Xamarin application, <a href="Development-Guide-Angular.html"><span class="text-primary">development guide</span></a> document for more
```
Comment (acjh) [inline doc/Getting-Started-Angular.html]: There's no `Getting-Started-Xamarin` — did you mean `Development-Guide-Xamarin`? (also: [here](https://github.com/aspnetzero/documents/pull/29/files/5518739bffa047a3eda136683430351008833e1a#diff-4da194f25b0fea35173718c38fbbd369R123))

Comment (hikalkan) [reply doc/Getting-Started-Angular.html]: Thank you @acjh for finding that. Fixing it.

Comment (ismcagdas) [reply doc/Getting-Started-Angular.html]: yes, there was a misunderstanding, I have fixed it, thanks 😄

Comment (acjh) [reply doc/Getting-Started-Angular.html]: 👍

**Pull request merged**